actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -94,13 +94,17 @@ def authenticate(request, &login_procedure)
       end
 
       def has_basic_credentials?(request)
-        request.authorization.present? && (auth_scheme(request).downcase == 'basic')
+        request.authorization.present? && (auth_scheme(request).downcase == 'basic') && !user_name_or_password_blank?(request)
       end
 
       def user_name_and_password(request)
         decode_credentials(request).split(':', 2)
       end
 
+      def user_name_or_password_blank?(request)
+        user_name_and_password(request).reject(&:blank?).empty?
```

Comment:
That would break some tests but `user_name_and_password(request).all?(&:blank?)` works and faster so I am changing it to that, thanks!